android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java

```diff
@@ -274,4 +287,56 @@ public void TestInvalidAddressFieldInput() {
         onView(withId(R.id.address_text_layout))
                 .check(matches(hasTextInputLayoutErrorText("")));
     }
+
+    @Test
+    public void testFutureDateNotPossible() {
+        DateTimeFormatter shortFormat = DateTimeFormatter.ofPattern("dd.MM.yy");
+        LocalDateTime now = LocalDateTime.now();
+        String today = shortFormat.format(now);
+
+        // Click on date icon
+        onView(withId(R.id.date_layout))
+                .perform(AddingFragmentTest.clickIcon(true));
+
+        // Click on edit icon to enter new date
+        onView(withId(com.google.android.material.R.id.mtrl_picker_header_toggle))
+                .perform(click());
+
+        // Replace current date with 31.12.2999
+        onView(withText(today))
+                .perform(replaceText("31.12.2999"));
+
+        // Close Keyboard
+        onView(withText("31.12.2999"))
+                .perform(closeSoftKeyboard());
```

Comment:
```suggestion
        closeSoftKeyboard();
```

This can be shortened